src/cmd/services/m3aggregator/config/aggregator.go

```diff
@@ -689,7 +690,10 @@ func (c flushManagerConfiguration) NewFlushManagerOptions(
 		opts = opts.SetMaxJitterFn(maxJitterFn)
 	}
 	if c.NumWorkersPerCPU != 0 {
-		workerPoolSize := int(float64(runtime.NumCPU()) * c.NumWorkersPerCPU)
+		workerPoolSize := int(math.Ceil(float64(c.NumProcessorsPerCPU * float64(runtime.NumCPU()))))
```

Comment:
sorry to keep going back and forth, can you break this up slightly?

Also if the `if workerPoolSize < 1` is redundant with the `math.Ceil` call isnt it?

Reply:
I broke it up a little more.

According to the yaml validation, it is technically possible to set NumWorkersPerCPU to 0.0 which would again result in 0 workers.  I'm not sure if you would rather change the validation or keep the `if workerPoolSize < 1` as a safeguard

Reply:
ah yeah duh forgot math.Ceil(0.0) is 0 not 1